OneSignal/OneSignal-iOS-SDK
# [feat] Include debug symbols (dSYM) in the SDK

# Description
## One Line Summary
Add debug symbols step to build script to include dSYM files in the SDK release.

## Details
- Only the first 2 commits are relevant. 
- The last 2 commits are for testing and will be removed after PR review.

### Motivation
- On Xcode 16, SDK consumers were getting warnings about missing dSYMS. While app validation still succeeds, the warnings can be alarming or a nuisance. On previous Xcode versions, the warnings were not generated.
- This may also now help us symbolicate crash reports better

### Scope
- Debug symbols only

# Testing

## Manual testing
**How to Test**
You can test using the steps under **Testing the SDK in another app without release** in our internal **OneSignal-iOS-SDK Release** Documentation.
- You can use this branch to add via Cocoapods
- SPM test branch is here: https://github.com/OneSignal/OneSignal-XCFramework/tree/dsym

I tested adding the SDK via both Cocoapods and Swift Package Manager. On both:
- app runs as expected
- the warnings went away

![before_with_warnings](https://github.com/user-attachments/assets/0c130fbc-44cc-47f0-a529-be003dde0ffc)

![after_without_warnings](https://github.com/user-attachments/assets/8fbc546e-d02f-4393-b6f3-ec10420c9464)

# Affected code checklist

# Checklist
## Overview

## Testing

## Final pass


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/OneSignal/OneSignal-iOS-SDK/1519)
